### DISCUSSION TITLE
The scroll bar disappears when TreeView set in multii-layer UserControls.

### DISCUSSION THREAD

**Discussion by textGamex** [AvaloniaUI/Avalonia#17708] [Q&A] [answered]
If I want to display a tree of folder by TreeView, which sets in multii-layer UserControls, the scroll bar won't show up while a part of TreeView surpasses the root window. I'm trying to bind the MaxHeight of TreeView to the Height of root Window, but it goes to crash. How could I do this correctly?

**Comment by timunie** [comment]
Don't put treeview into a SctollViewer or StackPanel, then hight should be contained automatically and internal Scrollviewer should work.

**Reply by textGamex** [reply]
I did not place the `TreeView` inside a `ScrollViewer` or `StackPanel`. The `TreeView` is located under the root node of the `UserControl`.

**Reply by timunie** [reply]
Please prepare a minimum sample and attach it here, that sounds unusual. Maybe a custom style?

**Reply by timunie** [reply]
https://github.com/AvaloniaUI/Avalonia/blob/master/src%2FAvalonia.Themes.Fluent%2FControls%2FTreeView.xaml#L30-L40

**Reply by textGamex** [reply]
mainWindow:

```xaml
<Window>
    <Grid RowDefinitions="Auto,Auto">
        <Menu
            x:Name="MainMenu"
            Grid.Row="0"
            SizeChanged="MainMenu_OnSizeChanged">
            <MenuItem Header="{x:Static lang:Resource.Menu_CharacterEditor}" />
            <MenuItem Header="{x:Static lang:Resource.Menu_Settings}" />
        </Menu>
        <ContentControl x:Name="MainContentControl" Grid.Row="1" />
    </Grid>
</Window>
```

`MainContentControl`:

```xaml
<UserControl>
    <Grid ColumnDefinitions="Auto,Auto">
        <ContentControl x:Name="SideBarControl" Grid.Column="0" />
        <ContentControl x:Name="WorkSpaceControl" Grid.Column="1" />
    </Grid>
</UserControl>
```

`SideBarControl.Content`:
```xaml
<UserControl>
    <TreeView
        ItemsSource="{Binding Items}"
        SelectionChanged="TreeView_OnSelectionChanged">
        <TreeView.ItemTemplate>
            <TreeDataTemplate x:DataType="models:SystemFileItem" ItemsSource="{Binding Children}">
                <TextBlock Text="{Binding Name}" />
            </TreeDataTemplate>
        </TreeView.ItemTemplate>
    </TreeView>
</UserControl>
```

**Reply by timunie** [reply, marked as answer]
RowDefinitiion Auto is the root cause. Auto will grow as much as needed and this can be infinite. Use * for the row with the TreeView, this will eat up remaining space. 

https://docs.avaloniaui.net/docs/reference/controls/grid

**Reply by textGamex** [reply]
Thank You
